### DISCUSSION TITLE
Apple clangd doesn't provide any documentation for libc functions, and :sh doesn't render man pages well

### DISCUSSION THREAD

**Discussion by butterthebuddha** [helix-editor/helix#11512] [Troubleshooting]
Hi folks,

I'm using Helix 24.07 to write some C code. Here's my LSP set up:
```
ani@Aniruddhs-MacBook-Pro ~$ hx --health c
Configured language servers:
  ✓ clangd: /usr/bin/clangd
Configured debug adapter: lldb-dap
Binary for debug adapter: 'lldb-dap' not found in $PATH
Configured formatter: None
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
ani@Aniruddhs-MacBook-Pro ~$ /usr/bin/clangd --version
Apple clangd version 15.0.0 (clang-1500.3.9.4)
Features: mac+xpc
Platform: x86_64-apple-darwin23.5.0; target=arm64-apple-darwin23.5.0
ani@Aniruddhs-MacBook-Pro ~$ hx -V
helix 24.7 (079f5442)
```

Unfortunately, Space-k doesn't produce any documentation for libc functions:
<img width="409" alt="Screenshot 2024-08-17 at 5 24 48 PM" src="https://github.com/user-attachments/assets/8020c08c-5836-47ec-91e6-46e1fd29f38b">

I tried to get around this by running `:sh man fork`, which mostly works, except the output doesn't render very well:
<img width="933" alt="Screenshot 2024-08-17 at 5 26 01 PM" src="https://github.com/user-attachments/assets/bf0feff8-612f-40de-8dec-53cc0a462cc4">

My guess this is due to overstriking: https://unix.stackexchange.com/questions/274658/purpose-of-ascii-text-with-overstriking-file-format

Are there any suggestions for working around this?

Thank you

**Comment by kirawi** [comment]
`man` is an interactive command which is not the intended use of `:sh`. I believe Apple strips their stdlib of documentation and I'm not aware of an easy workaround (I've heard some people install GCC/Clang stdlib for this instead). You may want to open `man` in a split.

**Reply by butterthebuddha** [reply]
> `man` is an interactive command

When you say interactive, you mean that `man` tries to use a pager, yes? I don't think `man` uses a pager if stdout isn't connected to a terminal (which should be the case when using `:sh`), and running `man fork | cat` does seem to suppress the pager in my terminal. 

As far as installing GCC/Clang: I may resort to that if I end up writing more C/C++ on my personal laptop, but I'd prefer to avoid having to do that for the light coding that I'm doing right now.

**Comment by butterthebuddha** [comment]
<img width="679" alt="Screenshot 2024-08-17 at 10 16 01 PM" src="https://github.com/user-attachments/assets/38a19aa9-ee43-4f4d-b0d7-b8208e972cec">

AI to the rescue! `:sh man fork | col -b` works beautifully.
